### ISSUE TITLE
Need help checking / updating setup guides

### ISSUE THREAD

**Issue by ofTheo** [openframeworks/openFrameworks#7625]
Hey all! 

Now 0.12.0 is out we'd love to get a little help checking and updating the current setup instructions for each platform. 

We're looking for volunteers to review a current setup guide.  eg: Xcode or Raspberry Pi and make suggestions / changes so it works with the current IDE and 0.12.0. 

Guides: 
 - [ ] https://openframeworks.cc/setup/xcode/
- [ ] https://openframeworks.cc/setup/linux-install/  ( for linux we would live platform specific instructions like Arch, Debian etc ) 
- [ ] https://openframeworks.cc/setup/emscripten/
- [ ] https://openframeworks.cc/setup/vs/
- [ ] https://openframeworks.cc/setup/msys2/
- [ ] https://openframeworks.cc/setup/raspberrypi/raspberry-pi-getting-started/ ( a section for non Pi boards would be welcome ) 

Drops/Adds: 
- [ ] Suggest dropping QT Creator: https://openframeworks.cc/setup/qtcreator/
- [ ] And adding a VS Code setup guide for platforms that have makefiles. 

If you are interested in helping with a guide just drop a note in this issue. 

The markdown files for the guides currently live here: 
https://github.com/openframeworks/ofSite/tree/master/content/setup 

A PR to the ofSite is ideal, but feel free to share a gist or by any means and we'll get it updated. 

Many thanks!

**Comment by danoli3** [comment]
Yeah absolutely. I'll do the android one as apart of the new updates next week

**Comment by oxillo** [comment]
@ofTheo 
I've started to update MSYS2 setup intructions. 
I'll PR once ready !

**Comment by pierrextardif** [comment]
I've tested with 0.11 on Pi zero2, and PI4, found slight misalignements, will check this one this week and report.

**Comment by Jonathhhan** [comment]
The Emscripten guide is still working for Linux and the Windows Subsystem for Linux. But I can check it again.
VS does not work for me yet: https://github.com/openframeworks/openFrameworks/issues/7484

**Comment by ofTheo** [comment]
Thanks everyone! 
Adding a note, that we should make clear minimum OS support ( ie: Ubuntu 18.04 / GCC 8.3, macOS 10.15 etc ).

**Comment by Jonathhhan** [comment]
For Emscripten I would add a note how to use multithreading...

**Comment by ofTheo** [comment]
Xcode is done thanks to @alptugan 🎉
https://github.com/openframeworks/ofSite/pull/807

**Comment by alptugan** [comment]
> Xcode is done thanks to @alptugan 🎉 [openframeworks/ofSite#807](https://github.com/openframeworks/ofSite/pull/807)

The image links seem to be broken. It looks fine in github, btw

**Comment by ofTheo** [comment]
ahh I think if you look at the way the translated versions are the files need to be linked like this:

```
![install dialog for the command line tools](/setup/xcode/install-dialog.png)
```
which corresponds to the file on the site:

```
https://openframeworks.cc/setup/xcode/install-dialog.png
```

so I think you would need to add: 
/setup/xcode to the image url ( which makes it not render in the markdown :P ) 

I just did a test to see: 
https://github.com/openframeworks/ofSite/commit/194842e43919c57f232b831450ee6e98473de4cd

**Comment by ofTheo** [comment]
Actually ignore that @alptugan - I think the files are just not getting synched over. 
I'll try and fix that.

**Comment by alptugan** [comment]
@ofTheo my bad :) because of retina display. I resize and PR

**Comment by diemildefreude** [comment]
Do you need help checking something on Windows?

**Comment by ofTheo** [comment]
Yes, please. 
I’d be good to double check the Visual Studio install guide and see if it needs to be brought up to date for VS 2022.

**Comment by diemildefreude** [comment]
OK, @ofTheo , here are some things I noticed:

**Install**

VS install is the same for 2022. Just that the components are newer:
![image](https://github.com/openframeworks/openFrameworks/assets/16056630/1d5e10a8-13b5-4c94-94d0-f6b02157926b)

When I open an sln file in the examples, I get "One or more projects in the solution were not loaded correctly" message. It looks like the directory was set to be the zip file that oF was originally downloaded in:
![image](https://github.com/openframeworks/openFrameworks/assets/16056630/10871ef9-2295-458b-b4f4-ca19d15e7d5f)

Using the Project Generator fixes this. 

However, the Project Generator also seems to generate projects with the v143 toolset, so you always have to retarget the solution. A newly generated project (or a preexisting solution from the examples folder) will give you the prompt for this, but when updating a project with the Project Generator (to add/remove addons), you need to right-click on the Solution in the Solution Explorer and choose Retarget Project:
![image](https://github.com/openframeworks/openFrameworks/assets/16056630/d49c8de5-f508-49ce-aefb-9ffa003011bc)

A section introducing the Project Generator seems like it'd be helpful. I guess bypassing it could avoid the the need to retarget(?), but it's otherwise a useful tool, so probably worth mentioning.

**Video Codecs**

Is K-Lite still needed on Windows with the new version? I thought there was a big change  with the way Windows video playback is handled in the new version...

**openFrameworks plugin for Visual Studio**
Opening Extensions>Manage Extensions and searching in the Online section reveals no openFrameworks plugin (I've also never used this iirc so it doesn't seem necessary)

![image](https://github.com/openframeworks/openFrameworks/assets/16056630/adfb6903-d2b3-4f26-aadc-7625d5676e59)

**Comment by ofTheo** [comment]
Thanks @diemildefreude ! 

> When I open an sln file in the examples, I get "One or more projects in the solution were not loaded correctly" message. It looks like the directory was set to be the zip file that oF was originally downloaded in:

Did you run the examples from the zip file by any chance? 
I think this could occur if you double click on the zip and open it like a folder instead of expanding it to a new directory. 

> However, the Project Generator also seems to generate projects with the v143 toolset, so you always have to retarget the solution. 

Yes, this is definitely intended behavior but maybe we could make sure it happens when you update a project it still asks to convert / retarget solution. That might just be a matter of removing a VS file in the project directory. 

> Video Codecs

That's right it's no longer needed for most video files. 

> openFrameworks plugin for Visual Studio

Definitely can be removed as well! 


Thank you!!

**Comment by diemildefreude** [comment]
> Did you run the examples from the zip file by any chance?
I think this could occur if you double click on the zip and open it like a folder instead of expanding it to a new directory.

@ofTheo I copied the oF folder from inside the zip into a folder outside of the zip file and then opened the solutions there.

**Comment by legendsovermyths** [comment]
@ofTheo, I've been actively using the library with VSCode and msys2 on Windows. If you need any assistance or have specific requirements, I'm more than willing to help. I can verify configurations and offer support, and I'm even open to writing guides if that would be beneficial. Just let me know how I can contribute!
